git/tagging-versions.md

```diff
@@ -0,0 +1,115 @@
+# Tagging versions
+
+Or, how to capture relevant information about deployments.
+
+## Rationale
+
+Without a plan, production deployments will happen and information about what
+was deployed will be scattered in various places. Hotfixes, which sometimes must
+cherry-pick specific commits out of the primary development timeline, add more
+complexity and confusion. The need almost invariably arises for a specific
+accounting of what was deployed. To that end, I propose the following best
+practice.
+
+## Levels of information
+
+Git already records commits, which include a summary message by the developer
+making the change. This level of information—both the actual files and lines
+changed as well as the summary messages—are the ground-level truth of what
+changed. It should be possible to easily list the commits specific to a version.
+
+The customer typically does not care about this level of information. Instead,
+customers track things at the level of tasks and features. Such features are
+typically tracked in Trello or another project management tool as "cards" (or
+similar). Cards have an associated identifying number. These card numbers should
+be explicitly listed in the information about a deployment as detailed below.
+
+## Versions
+
+Each deployment should have a version number associated with it. There is a
+one-to-one relationship between deployments and versions. Each version number is
+in the form `major.minor.patch`, e.g. 2.4.1, where the patch is optional and, if
+not specified, defined to be 0 (so `2.4` is a sort of shorthand for `2.4.0`).
+
+Each regularly scheduled deployment should result in a new minor version number.
+In most cases, the minor version number should be incremented. When the project
+craftsman deems the new version to be especially significant, the major version
```

Comment:
Or breaks dependencies in the case of [SemVer](http://semver.org/).  It's tricky to determine major version changes.  Wonder if we need any guidelines here.